coderd/autobuild/lifecycle_executor_test.go

```diff
@@ -1062,6 +1062,68 @@ func TestExecutorInactiveWorkspace(t *testing.T) {
 	})
 }
 
+func TestNotifications(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Dormancy", func(t *testing.T) {
+		t.Parallel()
+
+		// Setup template with dormancy and create a workspace with it
+		var (
+			ticker         = make(chan time.Time)
+			statCh         = make(chan autobuild.Stats)
+			notifyEnq      = testutil.FakeNotificationsEnqueuer{}
+			timeTilDormant = time.Minute
+			client         = coderdtest.New(t, &coderdtest.Options{
+				AutobuildTicker:          ticker,
+				AutobuildStats:           statCh,
+				IncludeProvisionerDaemon: true,
+				NotificationsEnqueuer:    &notifyEnq,
+				TemplateScheduleStore: schedule.MockTemplateScheduleStore{
+					GetFn: func(_ context.Context, _ database.Store, _ uuid.UUID) (schedule.TemplateScheduleOptions, error) {
+						return schedule.TemplateScheduleOptions{
+							UserAutostartEnabled: false,
+							UserAutostopEnabled:  true,
+							DefaultTTL:           0,
+							AutostopRequirement:  schedule.TemplateAutostopRequirement{},
+							TimeTilDormant:       timeTilDormant,
+						}, nil
+					},
+				},
+			})
+			admin   = coderdtest.CreateFirstUser(t, client)
+			version = coderdtest.CreateTemplateVersion(t, client, admin.OrganizationID, nil)
+		)
+
+		coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, admin.OrganizationID, version.ID)
+		userClient, _ := coderdtest.CreateAnotherUser(t, client, admin.OrganizationID)
+		workspace := coderdtest.CreateWorkspace(t, userClient, admin.OrganizationID, template.ID)
+		coderdtest.AwaitWorkspaceBuildJobCompleted(t, userClient, workspace.LatestBuild.ID)
+
+		// Stop workspace
+		workspace = coderdtest.MustTransitionWorkspace(t, client, workspace.ID, database.WorkspaceTransitionStart, database.WorkspaceTransitionStop)
+		_ = coderdtest.AwaitWorkspaceBuildJobCompleted(t, userClient, workspace.LatestBuild.ID)
+
+		// Wait for workspace to become dormant
+		ticker <- workspace.LastUsedAt.Add(timeTilDormant * 3)
+		<-statCh
+
+		// Check that the workspace is dormant
+		workspace = coderdtest.MustWorkspace(t, client, workspace.ID)
+		require.NotNil(t, workspace.DormantAt)
```

Comment:
.. unless it is flaky, then remove this check

Reply:
Since it is a background job, I think it is hard to find the exact time. Here is a diff between the value and the expectation:
```
expected: *time.Time(time.Date(2024, time.July, 24, 13, 37, 57, 414817000, time.UTC))
actual  : time.Time(time.Date(2024, time.July, 24, 13, 38, 56, 603858000, time.UTC))
```
I also don't think knowing if the dormant time is correct is helpful for this particular test.